### ISSUE TITLE
Public release 1.0.6

### ISSUE THREAD

**Issue by michaelhb** [michaelhb/superplot#21]
Creating a new thread for discussion of this release.

I have added superplot to the public pypi repository: https://pypi.python.org/pypi/superplot

The issue with appdirs / simpleyaml seems to be resolved. However I'm not able to try a clean install from scratch as I'm currently online via my phone (still waiting on telstra here :( ) - the downloads would destroy my quota. However I'll be borrowing a windows laptop later this week so I can try it out using the campus wifi.

**Comment by andrewfowlie** [comment]
I'm having problems related to pygtk on my Ubunutu computers.  I installed pygtk via aptitude. It works:

```
andrew@andrew-desktop:~$ python -c "import pygtk; print pygtk"
<module 'pygtk' from '/usr/lib/python2.7/dist-packages/pygtk.pyc'>
```

Though isn't there in `pip freeze | grep -i "pygtk"`. I cannot install it via pip (`* Building PyGTK using distutils is only supported on windows. *`).  Pip keeps trying to install it:

```
andrew@andrew-desktop:~$ pip install superplot
Requirement already satisfied (use --upgrade to upgrade): superplot in /usr/local/lib/python2.7/dist-packages
Requirement already satisfied (use --upgrade to upgrade): appdirs in /usr/local/lib/python2.7/dist-packages (from superplot)
Requirement already satisfied (use --upgrade to upgrade): prettytable in /usr/local/lib/python2.7/dist-packages (from superplot)
Requirement already satisfied (use --upgrade to upgrade): simpleyaml in /usr/local/lib/python2.7/dist-packages (from superplot)
Requirement already satisfied (use --upgrade to upgrade): numpy in /usr/lib/python2.7/dist-packages (from superplot)
Requirement already satisfied (use --upgrade to upgrade): matplotlib in /usr/local/lib/python2.7/dist-packages (from superplot)
Requirement already satisfied (use --upgrade to upgrade): scipy in /usr/lib/python2.7/dist-packages (from superplot)
Requirement already satisfied (use --upgrade to upgrade): pandas in /usr/lib/python2.7/dist-packages (from superplot)
Collecting pygtk (from superplot)
  Using cached pygtk-2.24.0.tar.bz2
    Complete output from command python setup.py egg_info:
    ********************************************************************
    * Building PyGTK using distutils is only supported on windows. *
    * To build PyGTK in a supported way, read the INSTALL file.    *
    ********************************************************************

  ----------------------------------------
  Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-LsPZ8_/pygtk
```

I can run `pip install --no-dependencies superplot` without problems. But then 

```
    andrew@andrew-desktop:~$ superplot_gui 
    Traceback (most recent call last):
      File "/usr/local/bin/superplot_gui", line 5, in <module>
        from pkg_resources import load_entry_point
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 3141, in <module>
        @_call_aside
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 3127, in _call_aside
        f(*args, **kwargs)
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 3154, in _initialize_master_working_set
        working_set = WorkingSet._build_master()
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 640, in _build_master
        ws.require(__requires__)
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 941, in require
        needed = self.resolve(parse_requirements(requirements))
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 828, in resolve
        raise DistributionNotFound(req, requirers)
    pkg_resources.DistributionNotFound: The 'pygtk' distribution was not found and is required by superplot
```

The pre-release candidate didn't have this problem (I've checked again now) `pip install --pre --index=https://testpypi/python.org/pypi_superplot --no-dependencies superplot` is successful (though I have a few issues if I remove --no-dependencies, see #18)

I can also install from source and run it without problems.

**Comment by michaelhb** [comment]
I've had similar issues. We may need to remove pygtk from the dependency list, or put in some platform specific logic for Linux /  Ubuntu. 

I'm currently out (at the police station actually - someone stole my number plates!) but I'll try to come up with a fix tomorrow.

**Comment by andrewfowlie** [comment]
OK, no worries. Good luck with the number plates, sounds strange.

**Comment by andrewfowlie** [comment]
I've just pushed a commit 020d8db to fix a bug: the median was missing from 2D plots. 

Maybe I said something incorrect in the past. The median isn't defined for a 2D distribution, but it's still useful to plot median(x) and median(y) on the x,y plane.

This isn't at all urgent. BTW I was in a meeting for a while this evening so will miss the talk today, hope it's an interesting discussion.

**Comment by michaelhb** [comment]
I've added a workaround for the pygtk issue. Setup.py now checks to see if pygtk can be imported, and only adds it to the dependency list if that fails. It'll still try, and fail to build pygtk on Ubuntu if it's not present, but once it's been obtained via apt-get, installation can proceed. And if the pip version of pygtk ever gets fixed, it'll just work.

I've also merged 020d8db into master and bumped the version to 1.0.9. EDIT: 1.0.10 actually.

The talk was good, although you didn't miss anything if you were at the first presentation. They also left out my favorite part (the "I'm having a ****ing scotch" line). Probably because there were loads of kids there :p

And I have new plates now. With anti-theft screws. Grr!
